elastic/kibana
# [Response Ops][Alerting] Refactor `ExecutionHandler` stage 2

Resolves https://github.com/elastic/kibana/issues/186534

## Summary

This PR splits the for-loop in the `ActionScheduler.run` function into the appropriate scheduler classes. Previously, each scheduler had a `generateExecutables` function that would return an array of executables and the `ActionScheduler` would loop through the array and convert the executable to a scheduleable action depending on whether it was a per-alert action, summary action or system action. This refactor renames `generateExecutables` into `getActionsToSchedule` and moves the logic to convert the executables into a schedulable action into the appropriate scheduler class.

## To Verify

Create some rules with per-alert and summary and system actions and verify they are triggered as expected.
